### PR TITLE
Add e2e-tests and api tests for doc-feature SP 2

### LINKED ISSUE #839: Add e2e/unit tests for document feature
- [x] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

### e2e-tests

test new structured doc table

### unit-test

- document_eventsourcing
- document_get
- document_share
- document_upload
- document_validation
- workflowitem_document_download
- public_key_eventsourcing
- public_key_get
- public_key_publish
- public_key_update (mark as unused) follow up issue "possibility to update private/public key"
- storage_service_url_eventsourcing
- storage_service_url_get
- storage_service_url_update
- storage_service_url
- private_key - (organization.ts- getPrivateKeyItem; getPrivateKey; publishPrivateKey)

follow up issue to #809

### PR DESCRIPTION
**Pull request by daniel-arnauer**
- [x] Add e2e-tests
- [x] Add API tests

e2e-tests
- test new structured doc table

unit-test:

- [x] document_eventsourcing
- [x] document_get
- [x] document_share
- [x] document_upload
- [x] document_validation
- [x] workflowitem_document_download
- [x] public_key_eventsourcing
- [x] public_key_get
- [x] public_key_publish
- [x] public_key_update (mark as unused) follow up issue "possibility to update private/public key"
- [x] storage_service_url_eventsourcing
- [x] storage_service_url_get
- [x] storage_service_url_update
- [x] storage_service_url
- [ ] private_key - (organization.ts- getPrivateKeyItem; getPrivateKey; publishPrivateKey)
- [x] secret_get

Closes #839

### CODE REVIEW

File: api/src/service/document_upload.ts [hunk 1]
```diff
@@ -23,7 +24,7 @@ export async function documentUpload(
   const uploadedDocumentResult = await Cache.withCache(conn, ctx, async (cache) => {
     return DocumentUpload.uploadDocument(ctx, serviceUser, requestData, {
       getAllDocumentInfos: async () => {
-        return await DocumentGet.getAllDocumentInfos(ctx, {
+        return DocumentGet.getAllDocumentInfos(ctx, {
           getDocumentsEvents: async () => {
```
Comment (daniel-arnauer) [inline api/src/service/document_upload.ts:28]: await is unnecessary - removed by eslint

File: api/src/service/domain/document/document_share.spec.ts [hunk 2]
```diff
@@ -49,67 +63,58 @@ describe("Share a document", async () => {
...
+        }),
+    });
+
+    assert.isTrue(Result.isErr(result));
   });
 
-  it("workflow_grant_permissions has to be set to share a document", async () => {
-    // TODO: THIS TEST FAILS
-    /*
+  it("User must belong to the same organization", async () => {
```
Comment (georgimld) [inline api/src/service/domain/document/document_share.spec.ts:112]: test should be deleted once this check is fixed in `document_share`

File: api/src/service/domain/organization/public_key_update.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const alice: ServiceUser = { id: "alice", groups: ["alice"] };
+
+const repository = {
+
+};
+
+describe("Public key: Update", async () => {
+
+
+    it("Updating a public key of an existing organization works", async () => {
```
Comment (georgimld) [inline api/src/service/domain/organization/public_key_update.spec.ts:26]: can this be tested already?

**Review by georgimld — CHANGES_REQUESTED** [review]

File: api/src/service/domain/document/document_download.spec.ts [hunk 4]
```diff
@@ -22,201 +22,217 @@ const projectId = "dummy-project";
...
-            getOffchainDocument: () => Promise.resolve(undefined),
-            getDocumentFromExternalStorage: () => Promise.resolve(uploadedExternalStorageDocument),
-            getDocumentInfo: () => Promise.resolve(documentInfoExternalStorage),
-        });
-
-        assert.isTrue(Result.isErr(offchainResult));
-        assert.isTrue(Result.isErr(storageResult));
-        assert.isTrue(Result.isErr(externalStorageResult));
+    console.log(result);
+    assert.isTrue(Result.isOk(result));
```
Comment (daniel-arnauer) [inline api/src/service/domain/document/document_download.spec.ts:173]: remove console log

**Review by georgimld — APPROVED** [review]

**Pull request merged**